src/main/perl/Fetch.pm

```diff
@@ -181,43 +173,48 @@ sub fetchProfile
 
     my $profile = $self->download("profile");
 
+    my $res;
     if (!defined($profile)) {
         $self->error("Failed to fetch profile $self->{PROFILE_URL}");
-        return undef;
-    }
-
-    return SUCCESS unless $profile;
-
-    local $SIG{__DIE__} = sub {
-        warn "Cleaning on die";
+    } elsif ("$profile" eq "0") {
+        # Have to do string comparison, because profile can be a FileReader or 0
+        $self->verbose("fetchProfile: nothing to do, ",
+                       "profile is not more recent than local cache (and no FORCE)");
```

Comment:
i'll fix the wording also in `retrieve`. wrt the 'no update', that's not true. it only is the timestamp on the remote server, not the content (https://github.com/quattor/CCM/blob/master/src/main/perl/Fetch/Download.pm#L104).